core/src/main/java/org/mskcc/cbio/oncokb/util/CacheUtils.java

```diff
@@ -375,7 +367,22 @@ public static Boolean containAlterations(Integer entrezGeneId) {
 
     public static void setAlterations(Gene gene) {
         if (gene != null && genes.contains(gene)) {
-            alterations.put(gene.getEntrezGeneId(), new HashSet<>(ApplicationContextSingleton.getAlterationBo().findAlterationsByGene(Collections.singleton(gene))));
+            List<Alteration> geneAlterations = ApplicationContextSingleton.getAlterationBo().findAlterationsByGene(Collections.singleton(gene));
+            if (!alterations.containsKey(gene.getEntrezGeneId())) {
+                alterations.put(gene.getEntrezGeneId(), new ArrayList<>());
+            }
```

Comment:
Do we need to initialize the map with an empty ArrayList value if it is immediately replaced by `geneAlterations` at line 377?

Reply:
Makes sense. I will update.